waspls/src/Wasp/LSP/ServerM.hs

```diff
@@ -0,0 +1,73 @@
+module Wasp.LSP.ServerM
+  ( ServerM,
+    ServerError (..),
+    Severity (..),
+    gets,
+    put,
+    modify,
+    logM,
+    lift,
+    catchError,
+    throwError,
+  )
+where
+
+import Control.Monad.Except (ExceptT, catchError, throwError)
+import Control.Monad.State.Strict (StateT)
+import qualified Control.Monad.State.Strict as StateT
+import Control.Monad.Trans (MonadIO (liftIO), lift)
+import Data.Text (Text)
+import Language.LSP.Server (LspT)
+import qualified System.Log.Logger as L
+import Wasp.LSP.ServerConfig (ServerConfig)
+import Wasp.LSP.ServerState (ServerState)
+
+type ServerM = ExceptT ServerError (StateT ServerState (LspT ServerConfig IO))
```

Comment:
Can you just have ServerM derive State stuff instead of reimplementing it? For that you will need newtype though. Don't bother with it if it doesn't mean much, I am mostly just interested what you think about it as alternative approach.

Reply:
I wanted to be able to attach more documentation about using lenses to the functions, but it might make more sense to put this in the module header and not have these aliases for state functions. What do you think?

Reply:
Aha that is a good reason!
I think it would probably make a bit more sense to put it in header, as you suggested. But, all together I don't think it is a big difference, so I am ok with both approaches.

Reply:
I've moved it to the header.